### ISSUE TITLE
State of the project

### ISSUE THREAD

**Issue by vladislav-kostin** [mzaks/ECS-Lang#3]
Hi. Do you mind if I ask here what are the current limitations of Entitas-Lang and what are your plans regarding further development? 

I'm still have not decided to commit to rewriting my current Unity project in Entitas, and I'm just trying it out so far. I certainly would switch almost certainly if Entitas would not have compile issues and would have stable blueprints, and I might still, and I would go with the Lang versions if it would be up to date with all the features, and would be more flexible in terms of file names and directories, and would support blueprints as well.

So is there any rough roadmap for the Entitas-Lang (ECS-Lang?)?

**Comment by sschmid** [comment]
Hi, great to hear you're considering Entitas.

@mzaks maintains Entitas-Lang. I'm currently trying to convince him that we need a json export so we can use the existing code generation infrastructure and don't have to maintain two codebases for that.

About Blueprints - don't use them, and don't plan to use them anytime soon :D I suggest having a custom solution for configuration. But if you really like blueprints and want to use them, it's really easy to add JSON blueprints which should be way easier to handle and migrate than the current binary blueprints. The main reason I haven't pushed them yet is because I didn't want to depend on a 3rd party lib. But you can totally do that. The json lib must be able to serialize `object` type.

Alternatively, there will be roslyn support soon. This approach doesn't generate partial abstract systems.

I just got the memo that maxim pushed Entitas-Lang
https://github.com/mzaks/Entitas-Lang

**Comment by sschmid** [comment]
Haha, though that's an issue on the Entitas Repo... sorry. Just saw it's ECS-Lang 🤦‍♂️

**Comment by mzaks** [comment]
Yeah this repo will be deprecated and https://github.com/mzaks/Entitas-Lang should be the new home. I will start working on it next week.

**Comment by vladislav-kostin** [comment]
@sschmid, lol. I've been thinking about it, and probably blueprints are not such a big deal, and even the 7 step component removal process too, but I do hope that you or someone else will implement my suggestion for fully utilizing the Unity's serializer: [https://waffle.io/sschmid/Entitas-CSharp/cards/58e937bdd90473001b081339](https://waffle.io/sschmid/Entitas-CSharp/cards/58e937bdd90473001b081339)

@mzaks, cool. I also hope Entitas-Lang will get more colors 😄
